LAB-251: ThorlabsLTS_hangs_on_some_coordinates_pseudo_irrational_numbers

samsnn opened the pull request:
Forced `MoveAbs()` to always round argument `pos` to 0.1 millimeters. This should prevent the stage from hanging after moving to certain irrational coordinates.

deepaksnn commented on test/LTS150M/test_position_rounding.jl:
```diff
@@ -0,0 +1,57 @@
+    @test measured_x_pos == 3u"mm"
+    @test measured_y_pos == 12u"mm"
+    @test measured_z_pos == 7u"mm"
+end
```
Add more test case with position value up to 3 or 4 decimal point to test the rounding functionality.
Add some boundary test case - close to home position (0mm) and close to end position (145mm/150mm).

deepaksnn commented on test/LTS150M/test_position_rounding.jl:
```diff
@@ -0,0 +1,57 @@
+# set stage home parameters
+home(ps)
+set_origin(ps)
+ThorlabsLTStage.set_limits(ps, (0u"mm", 0u"mm", 0u"mm"), (145u"mm", 145u"mm", 145u"mm"))
```
Can we increase the mx position close to 150mm?

samsnn replied:
I don't think we can use 150mm as the limit until a `find_home()` method is implemented. When using the 150mm upper limit, Julia prints the error: ERROR: Desired upper limit (150 mm) is greater than the device's max position (149.99542 mm).

samsnn commented:
* Added the helper `trunc_intrinsic_position()` for reducing intrinsic position precision. 
* Added absolute_position_tolerance argument to `pause()`.
* Cleaned up *test_position_rounding.jl*
